### ISSUE TITLE
How to composer require ?

### ISSUE THREAD

**Issue by socketman2016** [allegro/php-protobuf#78]
How can I require allegro/php-protobuf using composer?

In composer.json
```
{
    "require": {
        "pear/console_commandline": "^1.2"
    }
}
```

there is no package name

**Comment by hjagodzinski** [comment]
The `composer.json` you see is to specify the requirements for the `protoc` plugin that comes with `php-protobuf`, not for making php-protobuf installable by `composer`. The last time I checked the composer was not able to install a PHP extension (it means you need to install it before running composer install by hand, using an OS-specific package manager). There is ongoing project ([Pickle](https://github.com/FriendsOfPHP/pickle)) that aims to solve this issue. All you can do is to specify in your `composer.json` that you require a given extension to be installed. There is one more thing that @christophermancini [proposed](#52). We can make `php-protobuf` installable by composer but it would mean manual build and installation of an extension. Why? Because the only thing that composer installation can give us at this point is to put extension sources to the vendor directory. [Here](https://github.com/basho/riak-phppb-client#composer-install) you can find how Christopher handles it.

To make this extension installable on most of the operating systems is to publish this extension to the PECL. I have tried it once but unfortunately the project was not mature enough at this time and it's publication was rejected. It's time I will try once again.

**Comment by socketman2016** [comment]
solved #78

**Comment by socketman2016** [comment]
Solved #78
